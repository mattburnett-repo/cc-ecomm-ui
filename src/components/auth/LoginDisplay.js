
import { useContext } from 'react'
import { LoginMessageContext } from "../../features/auth/login";

export default function LoginDisplay ( props ) {
    const { handleClick, handleLogin } = props.handlers;

    const theApiUrl = process.env.REACT_APP_API_BASE_URL
    const theGoogleUrl = `${theApiUrl}/api/v1/auth/google`

    const message = useContext(LoginMessageContext)

    return (
        <div role="presentation" aria-label="login-display">
            <div role="presentation" aria-label="oauth">
                <div><a href={theGoogleUrl} aria-label="google-auth">Log in with Google</a></div>
                {/* <div><a href="/auth/github" aria-label="github-auth" onClick={handleClick}>Log in with GitHub</a></div>    */}
            </div>
            <div role="presentation" aria-label="login-display-form">
                {(message && <div role="presentation" aria-label='login-error-message'>{message}</div>)}
                <form onSubmit={handleLogin} method="post">
                    <div>
                        <label htmlFor="username">Username:</label>
                        <input id="username" aria-label="username" />
                    </div>
                    <div>
                        <label htmlFor="password">Password:</label>
                        <input id="password" type="password" aria-label="password" />
                    </div>
                    <div>
                        <button type="submit" aria-label="login">Log in</button>
                    </div>
                </form>  
            </div>
            <div role="presentation" aria-label="register">
                <label htmlFor="register">Don't have an account yet? </label>
                <a href='/register' id="register" aria-label="register-link">Register</a>
            </div>       
        </div>
    )
}