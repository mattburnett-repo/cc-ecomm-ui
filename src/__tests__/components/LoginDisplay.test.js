// 20211217: Wireframe https://wireframe.cc/6MXHyR

import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import LoginDisplay from '../../components/auth/LoginDisplay';

describe('<LoginDisplay /> component tests', () => {
    beforeEach(() => render(<LoginDisplay />));

    it('renders a google oauth link', () => {
        let theVal = screen.getByRole('link', { name: /google-auth/i});
        expect(theVal).toHaveTextContent('Log in with Google');
    });
    it('renders a github oauth link', () => {
        let theVal = screen.getByRole('link', { name: /github-auth/i});
        expect(theVal).toHaveTextContent('Log in with GitHub');
    });
    it('renders a username input', () => {
        screen.getByRole('textbox', { name: /username/i});
    });
    it('renders a password input', () => {
        screen.getByLabelText('Password:');
    });
    it('renders a login button', () => {
        screen.getByRole('button', { name: /login/i});
    });
    it('renders a Register link', () => {
        let theVal = screen.getByRole('link', { name: /register/i});
        expect(theVal).toHaveTextContent('Register');
    });

    it('renders a snapshot', () => {
        const tree = renderer.create(<LoginDisplay />).toJSON();
        expect(tree).toMatchSnapshot();
    });
}); // end component
