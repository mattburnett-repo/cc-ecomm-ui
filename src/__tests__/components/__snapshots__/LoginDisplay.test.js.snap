// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginDisplay /> component tests renders a snapshot 1`] = `
<div>
  <div>
    <a
      aria-label="google-auth"
      href="/auth/google"
    >
      Log in with Google
    </a>
  </div>
  <div>
    <a
      aria-label="github-auth"
      href="/auth/github"
    >
      Log in with GitHub
    </a>
  </div>
  <div>
    <label
      htmlFor="username"
    >
      Username:
    </label>
    <input
      aria-label="username"
      id="username"
    />
  </div>
  <div>
    <label
      htmlFor="password"
    >
      Password:
    </label>
    <input
      aria-label="password"
      id="password"
      type="password"
    />
  </div>
  <div>
    <button
      aria-label="login"
    >
      Log in
    </button>
  </div>
  <div>
    <label
      htmlFor="register"
    >
      Don't have an account yet? 
    </label>
    <a
      aria-label="register"
      href="/register"
      id="register"
    >
      Register
    </a>
  </div>
</div>
`;
